From 93ac1395f3735f18223ae4668933010a4a7989fd Mon Sep 17 00:00:00 2001
From: Gao Pan <pandy.gao@nxp.com>
Date: Fri, 2 Jun 2017 10:09:10 +0800
Subject: [PATCH 1872/5242] MLK-14999 arm: dts: imx7ulp: add ipg clk for i2c
 device node

commit  591018efb65efe854cd06a37af4ddc19c7d1e9ec from
https://source.codeaurora.org/external/imx/linux-imx.git

The lpi2c needs two clks, per clk and ipg clk. This patch adds ipg
clk for lpi2c device node.

Signed-off-by: Gao Pan <pandy.gao@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp.dtsi |   20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 9833c56..a8ebe44 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -229,8 +229,9 @@
 			compatible = "fsl,imx7ulp-lpi2c";
 			reg = <0x402B0000 0x10000>;
 			interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clks IMX7ULP_CLK_LPI2C4>;
-			clock-names = "ipg";
+			clocks = <&clks IMX7ULP_CLK_LPI2C4>,
+				 <&clks IMX7ULP_CLK_NIC1_BUS_DIV>;
+			clock-names = "per", "ipg";
 			assigned-clocks = <&clks IMX7ULP_CLK_LPI2C4>;
 			assigned-clock-parents = <&clks IMX7ULP_CLK_FIRC>;
 			assigned-clock-rates = <48000000>;
@@ -241,8 +242,9 @@
 			compatible = "fsl,imx7ulp-lpi2c";
 			reg = <0x402C0000 0x10000>;
 			interrupts = <GIC_SPI 35 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clks IMX7ULP_CLK_LPI2C5>;
-			clock-names = "ipg";
+			clocks = <&clks IMX7ULP_CLK_LPI2C5>,
+				 <&clks IMX7ULP_CLK_NIC1_BUS_DIV>;
+			clock-names = "per", "ipg";
 			assigned-clocks = <&clks IMX7ULP_CLK_LPI2C5>;
 			assigned-clock-parents = <&clks IMX7ULP_CLK_FIRC>;
 			assigned-clock-rates = <48000000>;
@@ -423,8 +425,9 @@
 			compatible = "fsl,imx7ulp-lpi2c";
 			reg = <0x40A40000 0x10000>;
 			interrupts = <GIC_SPI 36 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clks IMX7ULP_CLK_LPI2C6>;
-			clock-names = "ipg";
+			clocks = <&clks IMX7ULP_CLK_LPI2C6>,
+				 <&clks IMX7ULP_CLK_NIC1_BUS_DIV>;
+			clock-names = "per", "ipg";
 			assigned-clocks = <&clks IMX7ULP_CLK_LPI2C6>;
 			assigned-clock-parents = <&clks IMX7ULP_CLK_FIRC>;
 			assigned-clock-rates = <48000000>;
@@ -435,8 +438,9 @@
 			compatible = "fsl,imx7ulp-lpi2c";
 			reg = <0x40A50000 0x10000>;
 			interrupts = <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clks IMX7ULP_CLK_LPI2C7>;
-			clock-names = "ipg";
+			clocks = <&clks IMX7ULP_CLK_LPI2C7>,
+				 <&clks IMX7ULP_CLK_NIC1_BUS_DIV>;
+			clock-names = "per", "ipg";
 			assigned-clocks = <&clks IMX7ULP_CLK_LPI2C7>;
 			assigned-clock-parents = <&clks IMX7ULP_CLK_FIRC>;
 			assigned-clock-rates = <48000000>;
-- 
1.7.9.5

